src/Services/Importers/Sheet.php

```diff
@@ -40,12 +40,22 @@ public function handle()
 
     private function init(): void
     {
-        $this->iterator = $this->xlsx->rowIterator($this->sheet);
+        $this->iterator = $this->reader->rowIterator($this->sheet);
         $this->header = Sanitize::header($this->iterator->current());
         $this->rowLength = $this->header->count();
         $this->iterator->next();
     }
 
+    private function reader($import)
+    {
+        $file = Storage::path($import->file->path());
+
+        return match ($import->file->extension()) {
+            'csv' => new CSV($file),
+            default => new XLSX($file),
```

Comment:
'xlsx' => new XLSX($this->path),
default => throw new Enso Exception with message Unsupported import type